src/main/java/net/minestom/server/instance/EntityTrackerImpl.java

```diff
@@ -139,217 +130,103 @@ public void remove(@NotNull T removed) {
-        return Collections.unmodifiableList(chunkEntities);
-    }
-
-    @Override
-    public <T extends Entity> void nearbyEntitiesByChunkRange(@NotNull Point point, int chunkRange, @NotNull Target<T> target, @NotNull Consumer<T> query) {
-        final Long2ObjectSyncMap<List<Entity>> entities = targetEntries[target.ordinal()].chunkEntities;
-        if (chunkRange == 0) {
-            // Single chunk
-            final var chunkEntities = (List<T>) entities.get(CoordConversion.chunkIndex(point));
-            if (chunkEntities != null && !chunkEntities.isEmpty()) {
-                chunkEntities.forEach(query);
+    public synchronized <R extends Entity> @NotNull Stream<@NotNull R> selectEntity(@NotNull EntitySelector<R> selector, @NotNull Point origin) {
+        Stream<TrackedEntity> stream = switch (selector.target()) {
+            case ALL_ENTITIES -> idIndex.values().stream();
+            case ALL_PLAYERS -> playerIdIndex.values().stream();
+            case NEAREST_ENTITY -> {
+                final TrackedEntity nearest = findNearest(origin, false);
+                yield nearest != null ? Stream.of(nearest) : Stream.empty();
             }
-        } else {
-            // Multiple chunks
-            ChunkRange.chunksInRange(point, chunkRange, (chunkX, chunkZ) -> {
-                final var chunkEntities = (List<T>) entities.get(CoordConversion.chunkIndex(chunkX, chunkZ));
-                if (chunkEntities == null || chunkEntities.isEmpty()) return;
-                chunkEntities.forEach(query);
-            });
-        }
-    }
-
-    @Override
-    public <T extends Entity> void nearbyEntities(@NotNull Point point, double range, @NotNull Target<T> target, @NotNull Consumer<T> query) {
-        final Long2ObjectSyncMap<List<Entity>> entities = targetEntries[target.ordinal()].chunkEntities;
-        final int minChunkX = CoordConversion.globalToChunk(point.x() - range);
-        final int minChunkZ = CoordConversion.globalToChunk(point.z() - range);
-        final int maxChunkX = CoordConversion.globalToChunk(point.x() + range);
-        final int maxChunkZ = CoordConversion.globalToChunk(point.z() + range);
-        final double squaredRange = range * range;
-        if (minChunkX == maxChunkX && minChunkZ == maxChunkZ) {
-            // Single chunk
-            final var chunkEntities = (List<T>) entities.get(CoordConversion.chunkIndex(point));
-            if (chunkEntities != null && !chunkEntities.isEmpty()) {
-                chunkEntities.forEach(entity -> {
-                    final Point position = entriesByEntityId.get(entity.getEntityId()).getLastPosition();
-                    if (point.distanceSquared(position) <= squaredRange) query.accept(entity);
-                });
+            case NEAREST_PLAYER -> {
+                final TrackedEntity nearest = findNearest(origin, true);
+                yield nearest != null ? Stream.of(nearest) : Stream.empty();
```

Comment:
the nearest check should be done after the filter shouldnt it? Atleast in the command context, as thats how Mojang does it with their vanilla commands as you can filter out the closest entity and you then need to get the second closest or else it will always be empty.